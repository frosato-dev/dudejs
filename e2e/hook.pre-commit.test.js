import fs from "fs";
import path from 'path';
import { CLIENT_DIR } from "./constants";

const HOOK_PATH = path.join(CLIENT_DIR, '.git', 'hooks', 'pre-commit')

it("should have a pre-commit file", () => {
  const exits = fs.existsSync(HOOK_PATH)
  expect(exits).toBe(true);
});

it("should match snapshot", () => {
    const output = fs.readFileSync(HOOK_PATH, "utf8");
    expect(output).toMatchSnapshot()
});
