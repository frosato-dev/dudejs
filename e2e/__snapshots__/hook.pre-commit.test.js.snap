// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have a pre-commit file 1`] = `
"
#### DudeJS - START ####
npx dudejs staged
#### DudeJS - END ####
"
`;

exports[`should process staged files on pre-commit: formatted js file 1`] = `
"const hello = 'Hello'
"
`;

exports[`should process staged files on pre-commit: pre-commit error output 1`] = `
"
/tmp/dudejs-client/src/on-stage.js
  1:7  error  'hello' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

src/on-stage.js XXms
❌  Sorry dude, you have to fix your error(s) and commit again, be strong!
"
`;
